File: 프로그래머스/l2_카펫.py
def solution(brown, yellow):
    sqaure = brown + yellow
    
    can_make = []
    
    print(sqaure)
    for a in range(1, int(sqaure**(1/2)+1)):
        b, check = divmod(sqaure, a)
        if check == 0:
            can_make.append((b, a))
    print(can_make)
    
    for a, b in can_make:
        print(a, b, brown)
        if 2*(a+b-2) == brown:
            print(a, b)
            return [a,b]

solution(10,2)



# def solution(brown, yellow):
#     can_make = []
#     all_by = brown+yellow
#     for i in range(3, int((all_by)**0.5)+1):
#         if (all_by) % i ==0:
#             can_make.append((all_by//i, i))
#     for m, n in can_make:
#         if (m-2)*(n-2) == yellow:
#             return [m,n]

File: 프로그래머스/JAVA/l2_카펫.java
import java.util.ArrayList;
import java.util.Arrays;

public class l2_카펫 {
    public static void main(String[] args){
        int brown =8;
        int yellow =1;
        int[] answer = solution(brown ,yellow);
    }
    public static int[] solution(int brown, int yellow) {
        int square = brown + yellow;
        ArrayList<int[]> canMake = new ArrayList<>();

        for (int i=1; i<=Math.sqrt(square); i++){
            int b = square/i;
            int check = square%i;
            if (check==0)
                System.out.println(b+" "+i);
                canMake.add(new int[] {b, i});
        }
        int[] answer = {0,0};
        for(int[] numbers : canMake){
            if (2*(numbers[0]+numbers[1]-2)==brown){
                answer[0] = numbers[0];
                answer[1] = numbers[1];
            }

        }
        System.out.println(Arrays.toString(answer));
        return answer;
    }
}
